Communication tips for your open-source project (#762)
ropensci/roweb3: content/blog/2024-05-16-communication-tips-oss-project/index.Rmd

None of this takes a lot of time, but is good for your project's image.

## Required: Make sure you handle access smartly

This item is actually good for any account your project has (GitHub access rights, Slack admin rights, etc.).

For any platform requiring log-in / some sort of access rights, make sure everyone who needs access has it and that no one who doesn't need it still has it.

You might want to look into password management systems such as [1Password](https://1password.com/).

As the makeup of a development team changes over time, it might be wise to review access regularly, and to make it part of some sort of onboarding/offboarding cheklist.

## Required: have a space for private discussions

Although open-source development demands a lot to happen [in the open](https://producingoss.com/en/producingoss.html#avoid-private-discussions), having a safe space where team members can let off steam, discuss sensitive matters, or share pets' pictures, is very important.

> This could take the form of a Slack workspace, Discord server, etc.

Reviewer: ```suggestion
This could take such forms as a Slack workspace, Discord, Matrix, or Element server, or cutting edge options like [flat](https://try.flat.app) or [CQ2](https://cq2.co/).
```
(Just spicing it up a bit, and jumping at the chance to put CQ2 in there - I really want to create a need to spin that up!!)

Ideally, you should own the space, unless you can rely on an external partner (a funder? a larger cohalition of projects?) to keep providing it to you.

## Having a forum 

> For a small project, issue trackers might prove sufficient to handle bug reports, feature requests but also general Q&A.

Reviewer: ```suggestion
For a small project, issue trackers might be all you need to handle bug reports, feature requests, and general Q&A.
```

> Now, for bigger projects, your project might benefit from the creation and curation of a discussion forum.

Reviewer: ```suggestion
Bigger projects might nevertheless benefit from the creation and curation of a dedicated discussion forum.
```

You could use [Discourse](https://www.discourse.org/), [GitHub Discussions](https://docs.github.com/fr/discussions), etc. 
As with issue trackers, a challenge is making sure enough team or community members watch it.
Often, users of a forum will expect a faster turnaround than for an issue.

> 

Reviewer: I think this whole sub-section is redundant, and should be deleted, maybe just putting the sentiments of the final sentence into the previous sub-section on "space for private discussions"?

## Having a blog with an RSS feed

Compared to release notes, blog posts about new versions provide more of a narrative so can be easier to read.
They can still point users to the release notes for more information.

> The blog of an open-source project could also contain other types of posts such as a deep-dive on a feature, announcement of funding or call for contributions or monetary support, etc.

Reviewer: ```suggestion
The blog of an open-source project could also contain other types of posts such as a deep-dive on a feature, announcement of funding or call for contributions, or monetary support.
```
The ~~etc.~~ :bomb: strikes again!

When choosing a website builder, try to choose one that is free and either familiar for your project's team or easy enough to get acquainted with. 
Blog posts that are Markdown based are easier to write from release notes. 
Also make sure publishing a new blog post is not a complicated 100-step process or no one will want to write one.
You could choose to use [GitHub for a review and preview process of blog posts](https://blogguide.ropensci.org/).

If you do create a blog, make sure to also create an RSS feed for it.
With most static website generators this is either the default or available by turning on an option ([docs for Quarto](https://quarto.org/docs/websites/website-blog.html#rss-feed)).

Once your blog has an RSS feed, register it in relevant aggregators such as [R Weekly](https://github.com/rweekly/rweekly.org?tab=readme-ov-file#regular-r-posts-submit-your-rss-feed) in the R world.

## Having comments on blog posts

> If you choose to open comments on your blog posts, you can have that integrated with your project's forum!

Reviewer: ```suggestion
If you choose to open comments on your blog posts, make sure to integrate comments within your project's forum.
```

> It is possible with [Discourse](https://meta.discourse.org/t/embed-discourse-comments-on-another-website-via-javascript/31963) (which we used on this very blog), [GitHub Discussions via Giscus](https://giscus.app/fr) (easy to integrate with [Quarto](https://quarto.org/docs/output-formats/html-basics.html#commenting) among others).

Reviewer: ```suggestion
This is very easy with [Discourse](https://meta.discourse.org/t/embed-discourse-comments-on-another-website-via-javascript/31963) (which we used on this very blog), and [GitHub Discussions via Giscus](https://giscus.app/fr) (which are also easy to integrate with [Quarto](https://quarto.org/docs/output-formats/html-basics.html#commenting) among others).
```

> Integrating comments with your forum means you only need to watch one space, and also that blog posts might help make users aware of the forum.

Reviewer: ```suggestion
Integrating comments with your forum means you only need to watch one space, and also help connect readers of your blog posts to the forum.
```

## Having social media profiles

> Social media might be another place where to spread the word about your project and its updates, as well as to engage with users.

Reviewer: ```suggestion
Social media can be useful to spread the word about your project and its updates, and to engage with users.
```

> You might choose to make your social media "read-only" by stating clearly you do not have the resources to answer questions there for instace.

Reviewer: ```suggestion
You might choose to make your social media "read-only" by stating clearly you do not have the resources to answer questions there.
```

> Ideally, concentrate your social media usage on [nice platforms](/blog/2022/11/16/mastodon-en/), and platforms where your project's userbase usually congregates.

Reviewer: ```suggestion
Ideally, concentrate your social media usage on [nice platforms](/blog/2022/11/16/mastodon-en/), and platforms where your projects' users and community are likely to congregate.
```

## Conclusion

In this post, we shared some tips for communication for your open-source project.

> Use communication channels in accordance with your project's goals and resources: don't forget that aspect of your project's work but also don't spread your team too thin.

Reviewer: ```suggestion
Use communication channels in accordance with your project's goals and resources.
```
(I think that makes it more focussed and punchy.)
